tuna/connection.py

```diff
@@ -132,7 +155,57 @@ def is_alive(self):
         self.logger.error('Checking isAlive when channel does not exist')
       return not self.out_channel.exit_status_ready()
 
-  def exec_command_unparsed(self, cmd, timeout=SSH_TIMEOUT, abort=None):
+  def is_connected(self) -> bool:
+    """ Checks the status of connection """
+    status: bool = True
+    status = self.ssh.get_transport() is not None and \
+    self.ssh.get_transport().is_active() #type: ignore
+    return status
+
+  def connect(self, abort: Callable) -> None:
+    """Establishing new connecion"""
+    if not self.local_machine:
+      self.ssh_connect(abort)
+
+  def ssh_connect(self, abort: Any = None) -> bool:
+    """Establishing ssh connection"""
+    if not self.is_connected():
+      self.ssh = paramiko.SSHClient()
+      self.ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+      for ssh_idx in range(NUM_SSH_RETRIES):
+        if abort is not None and chk_abort_file(self.id, self.logger):
+          self.logger.warning('Machine %s aborted ssh connection', self.id)
+          return False
+
+        try:
+          self.ssh.connect(self.hostname,
+                           username=self.user,
+                           password=self.password,
+                           port=self.port,
+                           timeout=SSH_TIMEOUT,
+                           allow_agent=False)
+        except paramiko.ssh_exception.BadHostKeyException:
+          self.logger.error('Bad host exception which connecting to host: %s',
+                            self.hostname)
+        except (paramiko.ssh_exception.SSHException, socket.error):
+          retry_interval = randrange((int(SSH_TIMEOUT)))
+          self.logger.warning(
+              'Attempt %s to connect to machine %s (%s p%s) via ssh failed, \
+              sleeping for %s seconds', ssh_idx, self.id, self.hostname,
+              self.port, retry_interval)
+          sleep(retry_interval)
+        else:
+          self.logger.info(
+              'SSH connection successfully established to machine %s', self.id)
+          return True
+
+      self.logger.error('SSH retries exhausted machine: %s', self.hostname)
+      return False
+    return False
```

Comment:
I'm assuming this block was just moved and the only changes were to add return types and a couple variable types.

Reply:
redundant code is moved to a helper routin is_connected(), rest remains same.